### PR TITLE
PYTHON-4489 Make setup.py private

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -3,13 +3,21 @@ Changelog
...
+
 - The handshake metadata for "os.name" on Windows has been simplified to "Windows" to improve import time.
 - The repr of ``bson.binary.Binary`` is now redacted when the subtype is SENSITIVE_SUBTYPE(8).
 - A new asynchronous API with full asyncio support.
 
+Unavoidable breaking changes
+............................
+
+- Since we are now using ``hatch`` as our build backend, we no longer have a ``setup.py`` file
+  and require installation using ``pip``.
```
Comment (ShaneHarvey) [inline doc/changelog.rst:19]: This should be in 4.9 not 4.8, unless we are planning to backport PYTHON-4451.

**Review by ShaneHarvey — COMMENTED** [review]
What do you think about adding a setup.py that prints a message and exits with an error to help out users? Something like "PyMongo>=4.9 no longer supports building via setup.py, use `python -m pip install path/to/pymongo` instead."

File: doc/changelog.rst (hunk 1 above)
Comment (blink1073) [reply doc/changelog.rst:19]: It will be 4.8, PR is already open: https://github.com/mongodb/mongo-python-driver/pull/1663

**Comment by blink1073** [comment]
> What do you think about adding a setup.py that prints a message and exits with an error to help out users? 

Done

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**